### PR TITLE
Add build and test environment setup scripts, update developer guide

### PR DESCRIPTION
**Pull request by dsplaisted**
- Add scripts to set up Stage 0 (build) and Stage 2 (test) environments
- Update developer guide
- Delete old scripts which don't appear to be used or up-to-date

@johnbeisner @livarcocc @nguerrera @wli3 for review

### CODE REVIEW

File: scripts/cli-test-env.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ "$SOURCE" != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+
+REPO_ROOT="$( cd -P "$( dirname "$SOURCE" )/../" && pwd )"
+
+RID=linux-x64
```
Comment (nguerrera) [inline scripts/cli-test-env.sh]: How does this work on Mac?

File: Documentation/project-docs/developer-guide.md [hunk 2]
```diff
@@ -27,8 +27,8 @@ In order to build .NET Command Line Interface, you need the following installed
 ## Building/Running
 
 1. Run `build.cmd` or `build.sh` from the root depending on your OS. If you don't want to execute tests, run `build.cmd /t:Compile` or `./build.sh /t:Compile`. 
-  - To build the CLI in macOS Sierra, you need to set the DOTNET_RUNTIME_ID environment variable by running `export DOTNET_RUNTIME_ID=osx.10.11-x64`.
-2. Use `artifacts/{RID}/stage2/dotnet` to try out the `dotnet` command. You can also add `artifacts/{os}-{arch}/stage2` to the PATH if you want to use the build output when invoking `dotnet` from the current console.
+    - To build the CLI in macOS Sierra, you need to set the DOTNET_RUNTIME_ID environment variable by running `export DOTNET_RUNTIME_ID=osx.10.11-x64`.
```
Comment (nguerrera) [inline Documentation/project-docs/developer-guide.md]: This should not be necessary anymore

File: scripts/cli-test-env.sh (hunk 1 above)
Comment (wli3) [reply scripts/cli-test-env.sh]: bash shell behavior is different between mac and Linux? which specific part?

Comment (johnbeisner) [reply scripts/cli-test-env.sh]: Shouldn't the "RID" be determined programmatically?

Comment (johnbeisner) [reply scripts/cli-test-env.sh]: Something like this maybe:

RID=linux-x64 
if [ "$uname" = "Darwin" ]; then
    RID=osx-x64

**Comment by dsplaisted** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build
test Linux x64 Release Build
test OSX10.12 x64 Release Build
test Ubuntu16.04 x64 Debug Build

**Pull request merged**